core/src/main/java/io/questdb/griffin/engine/functions/math/GreatestNumericFunctionFactory.java

```diff
@@ -0,0 +1,211 @@
+    public String getSignature() {
+        return "greatest(V)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        final int[] counters = new int[11];
+
+        for (int i = 0; i < args.size(); i++) {
+            final Function arg = args.getQuick(i);
+            final int type = arg.getType();
+
+            switch (type) {
+                case ColumnType.FLOAT:
+                case ColumnType.DOUBLE:
+                case ColumnType.LONG:
+                case ColumnType.INT:
+                case ColumnType.SHORT:
+                case ColumnType.BYTE:
+                case ColumnType.TIMESTAMP:
+                    counters[type]++;
+                    continue;
+                default:
+                    if (arg.isNullConstant()) {
+                        return NullConstant.NULL;
+                    }
+                    throw SqlException.position(argPositions.getQuick(i)).put("unsupported type");
+            }
+        }
+
+        if (counters[ColumnType.DOUBLE] > 0) {
+            return new GreatestDoubleRecordFunction(new ObjList<>(args), argPositions);
+        }
+
+        if (counters[ColumnType.FLOAT] > 0) {
+            return new CastDoubleToFloatFunctionFactory()
+                    .newInstance(position,
+                            new ObjList<>(new GreatestDoubleRecordFunction(new ObjList<>(args), argPositions)),
+                            null, configuration, sqlExecutionContext);
+        }
+
+
+        if (counters[ColumnType.LONG] > 0) {
+            return new GreatestLongRecordFunction(new ObjList<>(args), argPositions);
+        }
+
+        if (counters[ColumnType.TIMESTAMP] > 0) {
+            return new CastLongToTimestampFunctionFactory()
+                    .newInstance(position,
+                            new ObjList<>(new GreatestLongRecordFunction(new ObjList<>(args), argPositions)),
+                            null, configuration, sqlExecutionContext);
+        }
+
+        if (counters[ColumnType.INT] > 0) {
+            return new CastLongToIntFunctionFactory()
+                    .newInstance(position,
+                            new ObjList<>(new GreatestLongRecordFunction(new ObjList<>(args), argPositions)),
+                            null, configuration, sqlExecutionContext);
+        }
+
+        if (counters[ColumnType.SHORT] > 0) {
+            return new CastLongToShortFunctionFactory()
+                    .newInstance(position,
+                            new ObjList<>(new GreatestLongRecordFunction(new ObjList<>(args), argPositions)),
+                            null, configuration, sqlExecutionContext);
+        }
+
+        if (counters[ColumnType.BYTE] > 0) {
+            return new CastLongToByteFunctionFactory()
```

Comment:
may be we can be more frugal here and bypass cast function factory call by skipping directly to cast function?

CastFactory always uses args.get(0), we don't need to create an extra obj list